source/server/server.h

```diff
@@ -141,6 +141,32 @@ class RunHelper : Logger::Loggable<Logger::Id::main> {
   Event::SignalEventPtr sig_hup_;
 };
 
+class ServerFactoryContextImpl : public Configuration::ServerFactoryContext {
+public:
+  explicit ServerFactoryContextImpl(Instance& server)
+      : server_(server), server_scope_(server_.stats().createScope("")) {}
+
+  Upstream::ClusterManager& clusterManager() override { return server_.clusterManager(); }
+  Event::Dispatcher& dispatcher() override { return server_.dispatcher(); }
+  const LocalInfo::LocalInfo& localInfo() const override { return server_.localInfo(); }
+  Envoy::Runtime::RandomGenerator& random() override { return server_.random(); }
+  Envoy::Runtime::Loader& runtime() override { return server_.runtime(); }
+  Stats::Scope& scope() override { return *server_scope_; }
+  Singleton::Manager& singletonManager() override { return server_.singletonManager(); }
+  ThreadLocal::Instance& threadLocal() override { return server_.threadLocal(); }
+  // TODO(lambdai): remove messageValidationVisitor from ServerFactoryContext
```

Comment:
Can this just be done as part of this change?

Reply:
Please allow me to address it in a follow up PR

Reply:
OK, but this really should be removed in the next PR in this space, as it's a problematic pattern, we now have multiple validation visitors being effectively delivered to the per-route filter factory method.